### PR TITLE
Remove the `@author` tag

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no 
| Tickets       |  no
| License       | MIT
| Doc PR        |

Highly controversial PR here, especially form me. I do love the `@author` tag. It is like a small diploma that says that I've done good. I remember the first time I got my name in the source code of a big projects. It really felt great. =)

But I also acknowledge that I currently have my name on a class that has been updated so many times that I dont really know how it works anymore. 

With this PR I propose that we should remove the author tag every major version. That means that if I add my name on a class, it will be there for 4-6 years until the class is no longer supported. It also means that if you see an author tag in 6.0, it means that the class has been introduced in this major version. 

I do think we should add `@author` tag on new classes. 

--------

One can argue that the author tag is useless because the information is in git. That is true, but the "diploma factor" is still important for some people.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
I'm personally not sold at all on this one.
I've quite often found the `@author` tag helpful in building a mental history of the project.
`git blame` is full of noise from CS changes and digging into it requires dedication.
`@author` don't, and that's their value.

Yes, sometimes they're not super accurate anymore, but I don't think that's a real issue that needs any fix.
While reviewing, we already reject non-legit additions.

TLDR, I see the drawbacks of this change but I don't see the benefits.

**Comment by Nyholm** [comment]
(php-cs-fixer is not fixing all classes. I'll do the remaining ~60 by hand later)

**Comment by Nyholm** [comment]
Sorry. I may not have been clear what issue I'm trying to address: 

The tag is often outdated and sometimes wrong after 2 or 3 years. After 6+ years it is just like a fossil. Sure fossils are interesting, but not if they are misleading.

**Comment by nicolas-grekas** [comment]
I'm not sure this is correct. From what I read in the code, they're quite accurate. Not 100% for sure, but eg compiler passes that mention `Johannes M. Schmitt` are correct, and they're more than 10 years old.

On the form component, removing `Bernhard Schussek` is also a loss to me. The tag is still accurate and will always remain, and it gives credit. Symfony is a community of human beings, and I think these are the little mentions that remind any readers about this.

**Comment by nicolas-grekas** [comment]
On the form component, removing `Bernhard Schussek` is also a loss to me. The tag is still accurate and will always remain, and it gives credit. Symfony is a community of human beings, and I think these are the little mentions that reminds the reader of the code about this.

**Comment by Nyholm** [comment]
Interesting. Those classes are exactly what I wound say are inaccurate. 

Yes, Johannes created a lot of great classes 10+ years ago. But almost all those classes has been heavily patched since. 

His "diploma" has also been around long enough and we cannot ping him if we have questions about his idea when created the class. 

For these 3 reasons, I think we should remove all his author tags. 

Obviously Johannes is just an example and I want to make a general rule that is easy to comply with.

**Comment by chalasr** [comment]
I agree with what Nicolas is saying here. 

These tags are important to some, and they basically don't harm.
If you can't use the tag to "blame" and get in touch with the author, then it's all about preserving credits for that person who originally created the file, which IMHO is totally fine since those credits are correct and deserved.

**Comment by StaffNowa** [comment]
Not sure why you want to do this, but not all possible to check ourself :) Sometimes maybe needs to have additional logins for some services :)

**Comment by lyrixx** [comment]
I agee with @nicolas-grekas here.

**Comment by dunglas** [comment]
I also agree with Nicolas. These tags are often helpful or/and funny, and never hurt.

**Comment by Nyholm** [comment]
I also think they are beneficial.  That is why I started the PR with: 

> I do love the `@author` tag. It is like a small diploma that says that I've done good. I remember the first time I got my name in the source code of a big projects. It really felt great. =)

I dont think we should stop using `@author` tags. 

However, I want to put more focus on the point Im trying to make that the **tags are often inaccurate, misleading and sometimes plain wrong**. 

I can find many more examples like #40649. Look at this class [from Messenger](https://github.com/symfony/symfony/blame/5.4/src/Symfony/Bridge/Doctrine/Messenger/DoctrineTransactionMiddleware.php). I originally created that class, but all the logic has been replaced since. The class even extends a different abstract class. 

So, my mission is to "*correct every wrong `@author` tags*". I could come up with a nice script and some rules to make it fair/accurate. Number of changes, percentage of code still created by the author etc. I could add a CI job to run this script every week to make sure the rules comply...

However, I prefer a simple rule: *Remove all author tags at every major release.*

**Comment by ro0NL** [comment]
i tend to agree with @Nyholm, at least assume ppl may also ask for removal and/or wont involve anyway. To me it represents the initial git author mostly.

There's also https://github.com/symfony/symfony/blob/5.4/.github/CODEOWNERS which may give better understanding who are involved where.

**Comment by nicolas-grekas** [comment]
I'm :+1: to remove any `@author` tag on `Tests` classes, they don't make any sense. That's one rule :)

If we're looking for rules, we should define when adding an `@author` should be rejected, and enforce it at the review stage IMHO.

And of course, one should be free to remove their `@author` tag if they feel like so. That's worth writing somewhere if that's not obvious.

(Code owners are another concept, tight to some GitHub features.)

**Comment by nicolas-grekas** [comment]
After discussing this with in the core-team, we decided to keep the status quo.

I would personally welcome removing the tag on test classes and we should think about those tags before merging during the review.

Thanks for the discussion.
